conan/api/subapi/install.py

```diff
@@ -70,7 +72,10 @@ def install_consumer(self, deps_graph, generators=None, source_folder=None, outp
 
         # The previous .set_base_folders has already decided between the source_folder and output
         if deploy or deploy_package:
-            base_folder = deploy_folder or conanfile.folders.base_build
+            base_folder = conanfile.folders.base_build
+            # Issue related: https://github.com/conan-io/conan/issues/16543
+            if deploy_folder:
+                base_folder = os.path.join(base_folder, deploy_folder)
```

Comment:
I am not sure this is correct. The ``deploy_folder`` if defined would be relative to the user ``cwd``, not the ``folders.base_build``

Reply:
I think that it's correct. Using a deployer without a `--deployer-folder`, the base folder was the `conanfile.folders.base_build`, so why should it not be `base + deployer-folder` using that parameter?

Reply:
I agree with that, but then it should be:

```python
            if deploy_folder:
                base_folder = os.path.join(os.getcwd(), deploy_folder)
```
I think it should always be an abs path. Wdyt?